libbpf-rs/src/map.rs

```diff
@@ -476,6 +476,26 @@ impl Map {
             ptr: std::ptr::null_mut(),
         })
     }
+
+    /// Attach a struct ops map
+    pub fn attach_struct_ops(&mut self) -> Result<Link> {
+        if self.map_type() != MapType::StructOps {
+            return Err(Error::InvalidInput(format!(
+                "Invalid map type ({}) for attach_struct_ops()",
+                self.map_type(),
+            )));
+        }
+
+        let ret = unsafe {
+            let p = libbpf_sys::bpf_map__attach_struct_ops(self.ptr);
```

Comment:
Unfortunately, I think this will crash in the case where the map is created by the userspace application (in the create call in line 428 above) because the ptr is null in those cases (we only get the fd). What are your thoughts on adding a check for whether self.ptr is null and erroring out in those cases?